api/src/main/java/io/strimzi/api/kafka/model/KafkaAssembly.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.strimzi.crdgenerator.annotations.Crd;
+import io.sundr.builder.annotations.Buildable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A description of a Kafka assembly, as exposed by the Strimzi Kafka CRD.
+ */
+@JsonDeserialize(
+        using = JsonDeserializer.None.class
+)
+@Crd(
+        apiVersion = KafkaAssembly.CRD_API_VERSION,
+        spec = @Crd.Spec(
+                names = @Crd.Spec.Names(
+                        kind = KafkaAssembly.RESOURCE_KIND,
+                        plural = KafkaAssembly.RESOURCE_PLURAL
+                ),
+                group = KafkaAssembly.RESOURCE_GROUP,
+                scope = "Namespaced",
+                version = KafkaAssembly.VERSION
+        )
+)
+@JsonPropertyOrder({"apiVersion", "kind", "metadata", "spec"})
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = true,
+        builderPackage = "io.strimzi.api.kafka.model"
+)
+public class KafkaAssembly extends CustomResource {
+
+
+    private static final long serialVersionUID = 1L;
+    public static final String VERSION = "v1alpha1";
+    public static final String RESOURCE_KIND = "Kafka";
+    public static final String RESOURCE_LIST_KIND = RESOURCE_KIND + "List";
+    public static final String RESOURCE_GROUP = "kafka.strimzi.io";
+    public static final String RESOURCE_PLURAL = "kafkas";
+    public static final String RESOURCE_SINGULAR = "kafka";
+    public static final String RESOURCE_NAME = RESOURCE_PLURAL + "." + RESOURCE_GROUP;
+    public static final String CRD_API_VERSION = "apiextensions.k8s.io/v1beta1";
+    public static final String CRD_NAME = RESOURCE_PLURAL + "." + RESOURCE_GROUP;
+
+    private String apiVersion;
+    private ObjectMeta metadata;
```

Comment:
It seems we are not using this field even because it's in the `CustomResource` base class and we are using getter/setter for that.

Reply:
This is actually working around an issue with sundrio, which is reported here: https://github.com/sundrio/sundrio/issues/104, but I hadn't realised until yesterday that overriding getMetadata() isn't enough, I actually have to have this field present too for it to generate a `withMetadata()` in the `KafkaAssemblyBuilder`. The issue is really about getting it to generate a `withNewMetadata()` method too.

Reply:
I didn't get the need for a field which is private and is not used. You are talking about overriding get/set Metadata which is something you are already doing but this field is not involved at all.

Reply:
Yeah, for whatever reason the field really needs to be there (even though it's unused) for sundrio to generate a build with `withMetadata()`. TBH that might be a bug in sundio, but since I need that method I'm willing to have the redundant field.